cluster-agent/controllers/argoproj.io/namespace_reconciler_test.go

```diff
@@ -748,4 +749,211 @@ var _ = Describe("Namespace Reconciler Tests.", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(gitopsEngineCluster).ToNot(BeNil())
+			Expect(created).To(BeTrue())
+
+			By("Create ClusterCredentials entry in DB.")
+
+			clusterCredentials := db.ClusterCredentials{
+				Clustercredentials_cred_id:  "test-creds-" + string(uuid.NewUUID()),
+				Host:                        "host",
+				Kube_config:                 "kube-config",
+				Kube_config_context:         "kube-config-context",
+				Serviceaccount_bearer_token: "serviceaccount_bearer_token",
+				Serviceaccount_ns:           "Serviceaccount_ns",
+			}
+			err = dbq.CreateClusterCredentials(ctx, &clusterCredentials)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create ManagedEnvironment entry in DB.")
+
+			managedEnvironment = db.ManagedEnvironment{
+				Managedenvironment_id: "test-env-" + string(uuid.NewUUID()),
+				Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+				Name:                  "my env",
+			}
+			err = dbq.CreateManagedEnvironment(ctx, &managedEnvironment)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create GitopsEngineInstance entry in DB.")
+
+			gitopsEngineInstance := db.GitopsEngineInstance{
+				Gitopsengineinstance_id: "test-id" + string(uuid.NewUUID()),
+				Namespace_name:          "test-ns-" + string(uuid.NewUUID()),
+				Namespace_uid:           "test-ns-" + string(uuid.NewUUID()),
+				EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+			err = dbq.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create Application entry in DB.")
+
+			application = db.Application{
+				Application_id:          "test-id" + string(uuid.NewUUID()),
+				Name:                    "test-name" + string(uuid.NewUUID()),
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+			err = dbq.CreateApplication(ctx, &application)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create ClusterAccess entry in DB.")
+
+			clusterAccess := db.ClusterAccess{
+				Clusteraccess_user_id:                   "test-user",
+				Clusteraccess_managed_environment_id:    managedEnvironment.Managedenvironment_id,
+				Clusteraccess_gitops_engine_instance_id: gitopsEngineInstance.Gitopsengineinstance_id,
+			}
+			err = dbq.CreateClusterAccess(ctx, &clusterAccess)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create Secret CR definition.")
+
+			secret = corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      argosharedutil.GenerateArgoCDClusterSecretName(managedEnvironment),
+					Namespace: gitopsEngineInstance.Namespace_name,
+				},
+			}
+		})
+
+		It("Should not create Operation for Secret, even if it is missing in Cluster, since ManagedEnvironment is created recently.", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("Create Secret CR in cluster.")
```

Comment:
Shouldn't we not create the secret on the cluster, since from the description of the test, it should be missing?